File: apps/backend/services/Business/Business.Presentation/Endpoints/Clients/UpdateClient.Request.cs
namespace Business.Presentation.Endpoints.Clients;

/// <summary>JSON body for PUT /clients/{id}.</summary>
public record UpdateClientRequest(
    string Name,
    string PhoneNumber,
    string Email,
    string? LocationAddress,
    double? LocationLatitude,
    double? LocationLongitude,
    string? Description);

File: apps/mobile/lib/core/features/clients/models/requests/update_client_request.dart
/// JSON body for PUT /clients/{id} — mirrors the Business API's UpdateClientRequest.
class UpdateClientRequest {
  final String name;
  final String phoneNumber;
  final String email;
  final String? locationAddress;
  final double? locationLatitude;
  final double? locationLongitude;
  final String? description;

  const UpdateClientRequest({
    required this.name,
    required this.phoneNumber,
    required this.email,
    this.locationAddress,
    this.locationLatitude,
    this.locationLongitude,
    this.description,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'phoneNumber': phoneNumber,
        'email': email,
        'locationAddress': locationAddress,
        'locationLatitude': locationLatitude,
        'locationLongitude': locationLongitude,
        'description': description,
      };
}
